helix-term/src/commands.rs

```diff
@@ -4328,13 +4326,19 @@ fn yank_joined_impl(editor: &mut Editor, separator: &str, register: char) {
     let (view, doc) = current!(editor);
     let text = doc.text().slice(..);
 
+    let separator = if separator.is_empty() {
+        doc.line_ending.as_str()
+    } else {
+        separator
+    };
+
     let selection = doc.selection(view.id);
     let selections = selection.len();
     let joined = selection
         .fragments(text)
         .fold(String::new(), |mut acc, fragment| {
             if !acc.is_empty() {
-                acc.push_str(separator);
+                acc.push_str(&helix_core::shellwords::unescape(separator));
```

Comment:
```suggestion
                acc.push_str(&shellwords::unescape(separator));
```

the shellwords module is `use`d above